esFrontLine/auth.py

```diff
@@ -27,26 +27,24 @@ def load_users(self, users):
         if not isinstance(users, list) or not len(users):
             return
 
-        def _valid_user(user):
-            # Check each user structure from settings
-            hawk = user.get('hawk')
-            resources = user.get('resources')
-            if hawk is None or resources is None:
-                return False
-
-            if not isinstance(resources, list) or not isinstance(hawk, dict):
-                return False
-
-            if hawk.keys() != ['algorithm', 'id', 'key']:
-                return False
-
-            logger.debug('Validated user {id}'.format(**user['hawk']))
-            return user
-
-        self.users = {
-            user['hawk']['id']: user
-            for user in filter(_valid_user, users)
-        }
+        self.users = {}
+        for index, user in enumerate(users):
+            try:
+                hawk = user.get('hawk')
+                resources = user.get('resources')
+                assert hawk is not None, 'Missing "hawk" setting in user config.'
+                assert resources is not None, 'Missing "resources" setting in user config.'
+                assert isinstance(resources, list), '"resources" must be JSON list'
+                assert len(resources) > 0, '"resources" cannot be empty'
+                assert isinstance(hawk, dict), '"hawk" must be a JSON dictionary'
+                assert hawk.keys() == ['algorithm', 'id', 'key'], \
```

Comment:
I believe this will not work on python3, where `keys()` returns a `set`

Reply:
Should i also support python 3 ? That would require some rewriting over `app.py`. I was thinking i could refactor it after landing the authentication code